README.md

```diff
@@ -8,15 +11,71 @@ If you want a module that specifically aims to be a complete and transparent int
 
 Initially this module was named Net::LDNS.
 
-IDN
-===
-If GNU libidn is installed when this module is compiled, it will be used to add a simple function that converts strings from Perl's internal encoding to IDNA domain name format. In order to convert strings from whatever encoding you have to Perl's internal format, use L<Encode>. If you need any kind of control or options, use L<Net::LibIDN>. The included function here is only meant to assist in the most basic case, although that should cover a lot of real-world use cases.
 
-Installation
-============
+## Features
+
+When installing from source, you can choose to enable or disable a number
+of optional features using command line options to the `perl Makefile.PL`
+commands.
+
+### IDN
+
+This feature is enabled by default.
+Disable with `--no-idn`.
+
+If the IDN feature is enabled, the GNU `libidn` library will be used to
+add a simple function that converts strings from Perl's internal encoding
+to IDNA domain name format.
+In order to convert strings from whatever encoding you have to Perl's
+internal format, use L<Encode>.
+If you need any kind of control or options, use L<Net::LibIDN>.
+The included function here is only meant to assist in the most basic case,
+although that should cover a lot of real-world use cases.
+
+> **Note:** The Zonemaster Engine test suite assumes this feature
+> is enabled.
+
+### Randomized capitalization
+
+This feature is disabled by default.
+Enable with `--randomize`.
+
+Enables compilation with randomzied capitalization.
+
+
+## Dependencies
+
+This module depends on the `openssl` library and has an optional
+dependency on the `libidn` (with the [IDN](#idn) feature, enabled by
+default) library.
+
+
+## Installation
+
 The recommended way to install Zonemaster::LDNS is to install it from CPAN as a dependency to Zonemaster::Engine. If you follow the [installation instructions for Zonemaster::Engine](https://github.com/dotse/zonemaster-engine/blob/master/docs/Installation.md) you will get all the prerequisites for Zonemaster::LDNS before installing it from CPAN.
 
-It is, of course, possible to install Zonemaster::LDNS (as all Zonemaster components) directly from a clone of the Git reposistory. There are, however, currently no direct instructions for that.
+To install Zonemaster LDNS from from sources, follow the instruction
+below.
+
+> **Note:** This procedure has additoinal compile-time dependencies on
+> `git`, `libtool` and `autoconf`
```

Comment:
I think there is some Perl modules required to run "perl Makefile.PL". Module::Install?

Reply:
Right. I obviously should have tested this on a *clean* machine.